pandas/tests/indexes/multi/test_setops.py

```diff
@@ -460,3 +460,12 @@ def test_intersection_different_names():
     mi2 = MultiIndex.from_arrays([[1], [3]])
     result = mi.intersection(mi2)
     tm.assert_index_equal(result, mi2)
+
+
+def test_union_empty_self_different_names():
```

Comment:
any reason not to put this with the other union tests above?

Reply:
The intersection test above is testing the Same for intersection. But we could move the union Test of course

Reply:
id prefer that, but not a big deal.

dont forget to fill in the GH ref

Reply:
Don't have a strong preference either. Moved it up a bit.

Added the gh ref